hyperspy/axes.py

```diff
@@ -2010,86 +2002,85 @@ def update_axes_attributes_from(self, axes,
             self.events.any_axis_changed.trigger(obj=self)
 
     def _update_attributes(self):
-        getitem_tuple = []
-        values = []
-        signal_axes = ()
-        navigation_axes = ()
         for axis in self._axes:
             # Until we find a better place, take property of the axes
             # here to avoid difficult to debug bugs.
             axis.axes_manager = self
-            if axis.slice is None:
-                getitem_tuple += axis.index,
-                values.append(axis.value)
-                navigation_axes += axis,
-            else:
-                getitem_tuple += axis.slice,
-                signal_axes += axis,
-        if not signal_axes and navigation_axes:
-            getitem_tuple[-1] = slice(axis.index, axis.index + 1)
 
-        self._signal_axes = signal_axes[::-1]
-        self._navigation_axes = navigation_axes[::-1]
-        self._getitem_tuple = tuple(getitem_tuple)
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(obj=self)
 
-        if len(self.signal_axes) == 1 and self.signal_axes[0].size == 1:
-            self._signal_dimension = 0
-        else:
-            self._signal_dimension = len(self.signal_axes)
-        self._navigation_dimension = len(self.navigation_axes)
+    def _on_size_changed(self):
+        self._update_attributes()
+        self.events.any_axis_changed.trigger(obj=self)
 
-        self._signal_size = (np.prod(self.signal_shape)
-                             if self.signal_shape else 0)
-        self._navigation_size = (np.prod(self.navigation_shape)
-                                 if self.navigation_shape else 0)
+    def _on_scale_changed(self):
+        self.events.any_axis_changed.trigger(obj=self)
 
-        self._update_max_index()
+    def _on_offset_changed(self):
+        self.events.any_axis_changed.trigger(obj=self)
```

Comment:
Remove `_on_size_change`, `_on_scale_changed` and `_on_offset_changed` and use a single callback, named `_on_any_axis_change` for example?